src/main/resources/db/migration/V1__create_tables.sql

```diff
@@ -0,0 +1,77 @@
+CREATE TYPE PHONE_TYPE as ENUM ('home', 'work', 'mobile');
+
+CREATE TYPE ROLE as ENUM ('associate', 'partner', 'senior');
+
+CREATE TYPE STATUS as ENUM ('new', 'assigned', 'on_hold',
+    'approved', 'canceled', 'declined');
+
+CREATE TABLE users
+(
+    id               bigserial
+        constraint PK_users PRIMARY KEY,
+    email            text not null
+        constraint UQ_users_email UNIQUE ,
+    cognito_username text not null
```

Comment:
Use text when you do not need to enforce a length constraint. It's suitable for storing long-form text, such as descriptions, comments, or JSON/XML data.

`text`: This data type can store strings of any size, up to the limit of 1 GB. There is no specific performance benefit to using text over varchar in PostgreSQL, as they are both treated the same internally.
`varchar(n)`: This is similar to the text type, but it can be restricted to a maximum length n. For example, varchar(36) will only store up to 36 characters. If you try to store a string longer than the specified limit, PostgreSQL will raise an error.

Reply:
Thank you fot such a deatailed code review.